### PR TITLE
T/13729 Table: Insert Column before and After fails if a cell vertical split exist

### LINKED ISSUE #591: Insert Column before and After fails if a cell vertical split exist
## Are you reporting a feature or a bug?

Bug

## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

http://dev.ckeditor.com/ticket/13729 (see also https://github.com/ckeditor/ckeditor-sdk/issues/164#issue-104653533).

## Provide detailed reproduction steps (if any)

1. Created a table with 4 columns and 3 rows.
1. Make a vertical split in the cell 2,2.
1. Do a column insert before in 4th column.

### Expected result
A new column is inserted between 3rd and 4th column.

### Actual result 
A new column will be created between 2nd and 3rd column.

## Other details

* Browser: All
* OS: All
* CKEditor version: 3.5.1 (reproduced in 4.7.1)
* Installed CKEditor plugins: Full

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](http://docs.ckeditor.com/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](http://docs.ckeditor.com/#!/guide/dev_tests) and
[how to create tests](http://docs.ckeditor.com/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

Closes #591.

### CODE REVIEW

File: tests/plugins/tabletools/manual/insertcolumnverticalsplit.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@bender-tags: 4.5.7, tc, tabletools, 13729
+@bender-ui: collapsed
+@bender-ckeditor-plugins: wysiwygarea, toolbar, table, tabletools, contextmenu
+
+1. Make a vertical split in the cell 2,2.
+2. Right click in the first cell in 4. column to insert a new column before it.
```
Comment (f1ames) [inline tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: Column inserted from cell 4,2 and 4,3 is inserted in the right place (between 3rd and 4th column). However, if you right click in the 4,1 cell and insert column it is placed on the beginning of the table as a first column.

Comment (f1ames) [inline tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: Well, it even happens without splitting any cell (step 1).

Comment (f1ames) [inline tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: Happens only in Chrome 58+ (OS X).

File: plugins/tabletools/plugin.js [hunk 2]
```diff
@@ -227,14 +227,17 @@
 		return null;
 	}
 
-	function getCellColIndex( cell, isStart ) {
+	function getCellColIndex( cell ) {
```
Comment (f1ames) [inline plugins/tabletools/plugin.js]: In the meantime, `tableselection` plugin have been created which also utilizes this function (passing `isStart=true`) in two places. The code there should be also adjusted. All `plugins/table*` unit tests are still passing which looks promising.

File: tests/plugins/tabletools/manual/insertcolumnverticalsplit.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+**Epected result:** A new column is inserted between 3rd and 4th column.
```
Comment (f1ames) [inline tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: Typo - should be **Expected**.

File: tests/plugins/tabletools/tabletools.js [hunk 4]
```diff
@@ -45,6 +45,9 @@
 			this.doTest( 'add-col-before-4', 'columnInsertBefore' );
 			this.doTest( 'add-col-before-multi', 'columnInsertBefore' );
 			this.doTest( 'add-col-before-multi2', 'columnInsertBefore' );
+
+			// (#13729)
```
Comment (f1ames) [inline tests/plugins/tabletools/tabletools.js]: We could refer here to GH issue which is #591.

File: tests/plugins/tabletools/manual/insertcolumnverticalsplit.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+@bender-tags: 4.5.7, tc, tabletools, 13729
```
Comment (f1ames) [inline tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: Please update tags here (CKEditor release tag and issue tag: `519, trac13729`)

File: tests/plugins/tabletools/manual/insertcolumnverticalsplit.md (hunk 1 above)
Comment (Comandeer) [reply tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: It's reproducible also in Safari. Additionally it's reproducible only if the selection is placed at the end of the cell. It seems to be connected with [known issue in Webkit/Blink](https://github.com/ckeditor/ckeditor-dev/pull/653#issuecomment-319969400) (it's nicely visible with `tableselection` plugin enabled; in such case cells 4.1 and 1.2 are selected after inserting new column). Probably very similar workaround as proposed for #577 could solve the issue. However this issue is not connected with the original one, so I'd move it to the dedicated issue.

Comment (Comandeer) [reply tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: Reported: #900

Comment (f1ames) [reply tests/plugins/tabletools/manual/insertcolumnverticalsplit.md]: 👍

**Pull request merged**